controlplane/kubeadm/controllers/controller.go

```diff
@@ -255,6 +267,11 @@ func (r *KubeadmControlPlaneReconciler) reconcile(ctx context.Context, cluster *
 	}
 
 	controlPlane := internal.NewControlPlane(cluster, kcp, ownedMachines)
+
+	// Aggregate the operational state of all the machines; while aggregating we are adding the
+	// source ref to the aggregate reason (reason@machine/name) so the problem can be easily tracked down to its source.
+	conditions.SetAggregate(controlPlane.KCP, controlplanev1.MachinesReadyCondition, ownedMachines.ConditionGetters(), conditions.AddSourceRef())
+
 	requireUpgrade := controlPlane.MachinesNeedingUpgrade()
 	// Upgrade takes precedence over other operations
 	if len(requireUpgrade) > 0 {
```

Comment:
```suggestion
	if len(needRollout) > 0 {
```

Can we change the name of these methods and variables to be `rollout` and not `upgrade`?

Reply:
done